code/modules/mob/living/life.dm

```diff
@@ -135,7 +135,8 @@
 	return 1
 
 /mob/living/proc/handle_vision()
-	client.screen.Remove(global_hud.blurry, global_hud.druggy, global_hud.vimpaired, global_hud.darkMask, global_hud.nvg, global_hud.thermal, global_hud.meson, global_hud.science)
+	if (client)

```

Comment:
This should be unnecessary.

The code right above should stop handle_vision from being called if there is no client in the body. I know that carbon/human/life() obeys by this.

If carbon/ayylmao/life() does not, then you should look into fixing some overrides you've got going on there. It should be unnecessary to perform this many sanity checks here.